interp/interp_test.go

```diff
@@ -1038,3 +1041,34 @@ func TestRunnerOpts(t *testing.T) {
 		})
 	}
 }
+
+func TestContext(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*3)
```

Comment:
Updated

I kept `time.WithTimeout` because I think it can not end before the select below, but I change that if you want. Changed time to 0.125 seconds.